test/data/s3/s3_data.py

```diff
@@ -0,0 +1,451 @@
+import os
+import sys
+import zlib
+import json
+from uuid import uuid4
+from hashlib import sha1
+from collections import namedtuple
+
+try:
+    # python2
+    from urlparse import urlparse
+except:
+    # python3
+    from urllib.parse import urlparse
+
+from metaflow.datatools.s3 import S3PutObject
+
+from metaflow.util import to_fileobj, to_bytes, url_quote
+
+import numpy
+
+from .. import s3client, S3ROOT
+
+BASIC_METADATA = {
+    #'no_meta': (None, None), # No metadata at all but going through the calls
```

Comment:
Is this supposed to remain commented out?

Reply:
That's a good point. Not sure. I can try uncommenting and seeing if it works well.

Reply:
I uncommented. Tests pass (you do have to re-upload your data if you already have it uploaded). THere are a few more tests to run but don't expect any issues.